rclcpp/src/rclcpp/node_interfaces/node_graph.cpp

```diff
@@ -215,6 +256,66 @@ NodeGraph::get_client_names_and_types_by_node(
   return services_and_types;
 }
 
+std::map<std::string, std::vector<std::string>>
+NodeGraph::get_action_client_names_and_types_by_node(
+    rclcpp::Node node,
+    const std::string & node_name,
+    const std::string & namespace_) const
+{
+  rcl_names_and_types_t action_names_and_types = rcl_get_zero_initialized_names_and_types();
```

Comment:
i think this needs to be finalized with `rcl_names_and_types_fini` before exiting from this function scope to avoid memory leak.